File: steps/javascript/update-pet.step.js
// steps/javascript/update-pet.step.js
const { update } = require('./js-store');

exports.config = { type:'api', name:'JsUpdatePet', path:'/js/pets/:id', method:'PUT', emits: [], flows: ['pets'] };
exports.handler = async (req) => {
  const b = req.body || {};
  const patch = {};
  if (typeof b.name === 'string') patch.name = b.name;
  if (['dog','cat','bird','other'].includes(String(b.species))) patch.species = b.species;
  if (Number.isFinite(b.ageMonths)) patch.ageMonths = Number(b.ageMonths);
  if (['available','pending','adopted'].includes(String(b.status))) patch.status = b.status;
  const updated = update(req.pathParams.id, patch);
  return updated ? { status:200, body:updated } : { status:404, body:{ message:'Not found' } };
};

File: src/python/update_pet_step.py
# steps/python/update_pet.step.py
config = { "type":"api", "name":"PyUpdatePet", "path":"/py/pets/:id", "method":"PUT", "emits": [], "flows": ["PyPetManagement"] }

async def handler(req, _ctx=None):
    try:
        import sys
        import os
        sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
        from src.services.pet_store import update
    except ImportError:
        return {"status": 500, "body": {"message": "Import error"}}
    pid = req.get("pathParams", {}).get("id")
    b = (req.get("body") or {})
    patch = {}
    if isinstance(b.get("name"), str): patch["name"] = b["name"]
    if b.get("species") in ["dog","cat","bird","other"]: patch["species"] = b["species"]
    if isinstance(b.get("ageMonths"), (int, float, str)):
        try: patch["ageMonths"] = int(b["ageMonths"])
        except Exception: pass
    if b.get("status") in ["new","available","pending","adopted","deleted"]: patch["status"] = b["status"]
    updated = update(pid, patch)
    return {"status": 200, "body": updated} if updated else {"status": 404, "body": {"message":"Not found"}}
